SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRemotePossession.cpp

```diff
@@ -31,25 +32,84 @@ void ASpatialTestRemotePossession::PrepareTest()
 		FinishStep();
 	});
 
-	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float DeltaTime) {
+	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, [this]() {
 		ATestPossessionPawn* Pawn =
 			GetWorld()->SpawnActor<ATestPossessionPawn>(LocationOfPawn, FRotator::ZeroRotator, FActorSpawnParameters());
 		RegisterAutoDestroyActor(Pawn);
 		FinishStep();
 	});
 
+	AddStep(TEXT("Clear Original Pawn Array"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		OriginalPawns.Empty();
+		FinishStep();
+	});
+
 	// Ensure that all Controllers are located on the right Worker
 	AddWaitStep(FWorkerDefinition::AllServers);
 
+	AddStep(TEXT("Save Original Pawns"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		for (ASpatialFunctionalTestFlowController* FlowController : GetFlowControllers())
+		{
+			ATestPossessionPlayerController* PlayerController = Cast<ATestPossessionPlayerController>(FlowController->GetOwner());
+			if (PlayerController != nullptr && PlayerController->HasAuthority() && PlayerController->GetPawn()->HasAuthority())
+			{
+				AddToOriginalPawns(PlayerController, PlayerController->GetPawn());
+				UE_LOG(LogTemp, Log, TEXT("Adding original pawn %s to array"), *PlayerController->GetPawn()->GetName());
+			}
+		}
+		FinishStep();
+	});
+
 	ATestPossessionPlayerController::ResetCalledCounter();
 }
 
+void ASpatialTestRemotePossession::AddCleanupSteps()
+{
+	AddStep(TEXT("Clean up the test"), FWorkerDefinition::AllServers, nullptr, nullptr, [this](float) {
```

Comment:
Why is this on tick?

Reply:
+1

Reply:
Everything was on tick by default, working to change that now